lib/client/session.go

```diff
@@ -644,6 +644,33 @@ func handleNonPeerControls(mode types.SessionParticipantMode, term *terminal.Ter
 	}
 }
 
+func handlePeerControls(term *terminal.Terminal, remoteStdin io.Writer) {
+	stdin := term.Stdin()
+	if term.IsAttached() {
+		// escape.NewReader is used to enable manual disconnect sequences as those supported
+		// by tsh. These can be used to force a client disconnect since CTRL-C is merely passed
+		// to the other end and not interpreted as an exit request locally
+		stdin = escape.NewReader(stdin, term.Stderr(), func(err error) {
+			log.Debugf("escape.NewReader error: %v", err)
+
+			switch err {
+			case escape.ErrDisconnect:
+				fmt.Fprint(term.Stderr(), "\r\nDisconnected\r\n")
+			case escape.ErrTooMuchBufferedData:
+				fmt.Fprint(term.Stderr(), "\r\nremote peer may be unreachable, check your connectivity\r\n")
+			default:
+				fmt.Fprint(term.Stderr(), "\r\nunknown error\r\n")
```

Comment:
Maybe include err.Error() in the message at least?
```suggestion
				fmt.Fprint(term.Stderr(), "\r\nUnknown error\r\n")
```